IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadataCacheItem.m

```diff
@@ -187,6 +192,7 @@ - (BOOL)isEqualToItem:(MSIDAccountMetadataCacheItem *)item
     result &= (!self.clientId && !item.clientId) || [self.clientId isEqualToString:item.clientId];
     result &= ([_accountMetadataMap isEqualToDictionary:item->_accountMetadataMap]);
     result &= (!self.principalAccountId && !item.principalAccountId) || [self.principalAccountId isEqual:item.principalAccountId];
+    result &= (!self.principalAccountEnvironment && !item.principalAccountEnvironment) || [self.principalAccountEnvironment isEqual:item.principalAccountEnvironment];
```

Comment:
I think this should be isEqualToString for NSString objects.

Reply:
Addressed, thanks :)